common/output.h

```diff
@@ -55,11 +70,7 @@ extern short logg_syslog;
 int logg_facility(const char *name);
 #endif
 
-#ifdef __GNUC__
-void mprintf(const char *str, ...) __attribute__((format(printf, 1, 2)));
-#else
-void mprintf(const char *str, ...);
-#endif
+void mprintf(int loglevel, const char *str, ...);
```

Comment:
I hadn't considered before that we lose the `__attribute__((format(printf, 1, 2))` format string checks, but honestly it's worth it to me to fix this bug and get rid of the hacky `buff[0]` checks and un-intuitive log-level special characters.  👍

Reply:
Oh well, I think we don't need to lose them. We can shift the args, then it'll work even with new loglevel (:

Reply:
See [4448c09](https://github.com/Cisco-Talos/clamav/pull/445/commits/4448c092c83d1297706e12a6c029677d921e4ff1)